core/domain/story_services.py

```diff
@@ -535,3 +536,83 @@ def save_story_summary(story_summary):
     )
 
     story_summary_model.put()
+
+
+def get_completed_node_ids(user_id, story_id):
+    """Returns the ids of the nodes completed in the story.
+
+    Args:
+        user_id: str. ID of the given user.
+        story_id: str. ID of the story.
+
+    Returns:
+        list(str). List of the node ids completed in  story.
+    """
+    progress_model = user_models.StoryProgressModel.get(user_id, story_id)
+
+    return progress_model.completed_nodes if progress_model else []
+
+
+def get_node_ids_completed_in_stories(user_id, story_ids):
+    """Returns the ids of the nodes completed in each of the stories.
+
+    Args:
+        user_id: str. ID of the given user.
+        story_ids: list(str). IDs of the stories.
+
+    Returns:
+        list(list(str)). List of the node ids completed in each
+            story.
+    """
+    progress_models = user_models.StoryProgressModel.get_multi(
+        user_id, story_ids)
+
+    node_ids_completed_in_stories = []
+
+    for progress_model in progress_models:
+        if progress_model:
+            node_ids_completed_in_stories.append(
+                progress_model.completed_nodes)
+        else:
+            node_ids_completed_in_stories.append([])
+
+    return node_ids_completed_in_stories
+
+
+def get_completed_nodes_in_story(user_id, story_id):
+    """Returns nodes that are completed in a story
+
+    Args:
+        user_id: str. The user id of the user.
+        story_id: str. The id of the story.
+
+    Returns:
+        list(StoryNode): The list of the story nodes that the user has
+        completed.
+    """
+    story = get_story_by_id(story_id)
+    completed_nodes = []
+
+    completed_node_ids = get_completed_node_ids(user_id, story_id)
+    for node in story.story_contents.nodes:
+        if node.id in completed_node_ids:
+            completed_nodes = [node.to_dict()]
```

Comment:
Are there tests for this? Since, I think you have to do list.extend() here.
Also, why are you converting to dict? In the docstring, it is written list(StoryNode). I think domain objects are fine.

Reply:
I was thinking of returning a more human readable format to the controller layer, but I guess that could b done in the summary services. Will change this.

Reply:
Also, tests added.